neqo-transport/src/connection/mod.rs

```diff
@@ -1908,7 +1912,7 @@ impl Connection {
                         // a packet on a new path, we avoid sending (and the privacy risk) rather
                         // than reuse a connection ID.
                         let res = if path.borrow().is_temporary() {
-                            assert!(!cfg!(test), "attempting to close with a temporary path");
+                            debug_assert!(!cfg!(test), "attempting to close with a temporary path");
                             Err(Error::InternalError)
```

Comment:
Bubbling  `Error::InternalError` up multiple levels will have it loose semantic context, in other words, we won't know what exactly failed from the outside.

I suggest either logging here, or, long term, we might want to consider embracing [Rust's Error backtraces](https://doc.rust-lang.org/nightly/std/error/trait.Error.html#method.provide) via `thiserror` or [`snafu`](https://docs.rs/snafu/latest/snafu/). Though at least the former still depends on Rust Nightly.